### DISCUSSION TITLE
Compact Mode: OSC Elements

### DISCUSSION THREAD

**Discussion by Samillion** [Samillion/ModernZ#256] [Ideas]
I've been focused a lot on the compact mode layout for ModernZ, requested by @houssemko and discussed previously with @Xurdejl and @Keith94 in https://github.com/Samillion/ModernZ/discussions/209#discussioncomment-11248722

I'm liking it a lot the more I apply it. I don't have previews ready yet, but I might make it default (with both options available).

**Compact OSC Sketch:**
```
[title]
[time] ==================[seekbar]================== [time]
[play control+volume]                 [track+extra buttons]
```

Additionally, @guidocella has been doing great work with console/select, and `show-properties` is tempting me so much.

In essence, global menu for playlist, chapters, tracks, edition:

![XRwB avif](https://github.com/user-attachments/assets/cf07970a-6a0a-46e3-af24-fe83248259ac)

Which would be great for compact mode, combining `Playlist`, `Audio Track` and `Subtitle Track` into a single button.

## Proposed changes:
- Place timestamps in the same line as seekbar (to reduce deadspace)
- Remove "jump buttons" from compact mode completely
- Combine all menus into a single button (Playlist, chapters, subtitle tracks, audio tracks..etc)
- [Hesitant] Add chapter buttons on far right or left in [title] line
  - `[chapter buttons] [title]` or `[title]-----------[chapter buttons]`

**Comment by Keith94** [comment]
>Place timestamps in the same line as seekbar (to reduce deadspace)

It should probably handle dynamic times and milliseconds too, otherwise it can overlap the seekbar (as seen in [this fork](https://github.com/Qiyue0726/awesome-osc))

>Remove "jump buttons" from compact mode completely

Fine with that since it can be achieved with keybinds anyway.

>Combine all menus into a single button (Playlist, chapters, subtitle tracks, audio tracks..etc)

Would be nice to have or even as a context menu.

>[Hesitant] Add chapter buttons on far right or left in [title] line

How would that work if you have the title disabled.. use the window title instead? Hmm.

**Reply by nekoxuee** [reply]
> > Place timestamps in the same line as seekbar (to reduce deadspace)
> 
> It should probably handle dynamic times and milliseconds too, otherwise it can overlap the seekbar (as seen in [this fork](https://github.com/Qiyue0726/awesome-osc))

I have an old branch in my config with inline timestamps, https://github.com/Xurdejl/mpv-config/tree/feat/timecode-seekbar-inline in case it helps.
<br>

> Remove "jump buttons" from compact mode completely

Personally, I never use them, only the hotkeys
<br>

> Combine all menus into a single button (Playlist, chapters, subtitle tracks, audio tracks..etc)

This is a cool feature, even for the regular version.

**Reply by Samillion** [reply]
Fully agreed. Thank you guys very much for the references, it helps a lot. In a few days I'll post a working version for testing and adjustment.

**Comment by Samillion** [comment]
Check this out: https://github.com/mpv-player/mpv/pull/15441

Now that's another reason that makes the "one button" menu even more awesome.

**Comment by Samillion** [comment]
Follow up here please: https://github.com/Samillion/ModernZ/issues/208#issuecomment-2562212179
